### PR TITLE
Remove legacy log sharing & associated code & tests

### PR DESCRIPTION
**Pull request by mikejholly**

### CODE REVIEW

File: cmd/earthly/app/before.go [hunk 1]
```diff
@@ -23,69 +23,74 @@ import (
 )
 
 func (app *EarthlyApp) before(cliCtx *cli.Context) error {
-	if app.BaseCLI.Flags().EnableProfiler {
+	flags := app.BaseCLI.Flags()
```
Comment (alexcb) [inline cmd/earthly/app/before.go:26]: nice change :)

File: cmd/earthly/app/before.go [hunk 2]
```diff
@@ -23,69 +23,74 @@ import (
...
-			disableOngoingUpdates, app.BaseCLI.Flags().LogstreamDebugFile, app.BaseCLI.Flags().BuildID, execStatsTracker, app.BaseCLI.Flags().GithubAnnotations)
-		app.BaseCLI.SetLogbusSetup(newSetup)
-		if err != nil {
-			return errors.Wrap(err, "logbus setup")
-		}
+	var execStatsTracker *execstatssummary.Tracker
+	if flags.ExecStatsSummary != "" {
+		execStatsTracker = execstatssummary.NewTracker(flags.ExecStatsSummary)
+	}
+	newSetup, err := logbussetup.New(
```
Comment (alexcb) [inline cmd/earthly/app/before.go:57]:
```suggestion
	theOnlySetup, err := logbussetup.New(
```

or maybe just `setup` :laughing:

File: cmd/earthly/app/before.go (hunk 1 above)
Comment (mikejholly) [reply cmd/earthly/app/before.go:26]: Thanks. It's a little easier on the eyes :slightly_smiling_face:

File: cmd/earthly/app/before.go (hunk 2 above)
Comment (mikejholly) [reply cmd/earthly/app/before.go:57]: Lol. I don't if that name means "the setup using the new approach" or just "the newly created setup variable". I'll clarify it, nonetheless.

Comment (alexcb) [reply cmd/earthly/app/before.go:57]: yeah `setup` seems like a bad name; I too don't really know what it means at first glance.

Comment (mikejholly) [reply cmd/earthly/app/before.go:57]: Sorry, I meant the existing name `newSetup`.

File: cmd/earthly/app/before.go [hunk 3]
```diff
@@ -23,69 +23,74 @@ import (
...
+		flags.DisplayExecStats,
+		envutil.IsTrue("FORCE_COLOR"),
+		envutil.IsTrue("NO_COLOR"),
+		app.BaseCLI.Flags().InteractiveDebugging,
+		flags.LogstreamDebugFile,
+		flags.BuildID,
+		execStatsTracker,
+		flags.GithubAnnotations,
+	)
+	app.BaseCLI.SetLogbusSetup(newSetup)
```
Comment (alexcb) [inline cmd/earthly/app/before.go:71]: something seems off here:

```
	theOnlySetup, err := logbussetup.New(
	....
	)
	app.BaseCLI.SetLogbusSetup(newSetup)  <---- Shouldn't this happen after the err != nil check?
	if err != nil {
		return errors.Wrap(err, "logbus setup")
	}
```

Comment (mikejholly) [reply cmd/earthly/app/before.go:71]: That does seem odd. The previous commit has the same order, but it seems incorrect.

Comment (alexcb) [reply cmd/earthly/app/before.go:71]: yeah I see that, still seems odd -- I think fixing it is worthwhile.

**Review by alexcb — APPROVED** [review]

**Comment by mikejholly** [comment]
@alexcb @vladaionescu Ok to merge?

**Review by alexcb — APPROVED** [review]
merge away!

**Pull request merged**